amundsen_application/static/js/components/common/TagsList/index.tsx

```diff
@@ -5,57 +5,153 @@ import * as React from 'react';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 
-import './styles.scss';
-
 import ShimmeringTagListLoader from 'components/common/ShimmeringTagListLoader';
-
 import TagInfo from 'components/Tags/TagInfo';
 import { Tag } from 'interfaces';
 
 import { GlobalState } from 'ducks/rootReducer';
 import { getAllTags } from 'ducks/tags/reducer';
 import { GetAllTagsRequest } from 'ducks/tags/types';
-import { getCuratedTags, showAllTags } from 'config/config-utils';
+import { getCuratedTags } from 'config/config-utils';
+
+import {
+  POPULAR_TAGS_TITLE,
+  CURATED_TAGS_TITLE,
+  POPULAR_TAGS_NUMBER,
+  OTHER_TAGS_TITLE,
+} from './constants';
+import './styles.scss';
 
 export interface StateFromProps {
   curatedTags: Tag[];
   otherTags: Tag[];
   isLoading: boolean;
 }
 
+interface OwnProps {
+  /* determine if we only want curated/popular tags with
+  'Browse more tags' link or the entire tags list */
+  shortTagList: boolean;
+}
+
 export interface DispatchFromProps {
   getAllTags: () => GetAllTagsRequest;
 }
 
-export type TagsListProps = StateFromProps & DispatchFromProps;
+export type TagsListProps = StateFromProps & DispatchFromProps & OwnProps;
 
 export class TagsList extends React.Component<TagsListProps> {
   componentDidMount() {
     this.props.getAllTags();
   }
 
+  getTagsByUsage(tagArray: Tag[]) {
+    return tagArray
+      .sort((a, b) => {
+        return a.tag_count - b.tag_count;
+      })
+      .reverse();
+  }
+
   generateTagInfo(tagArray: Tag[]) {
     return tagArray.map((tag, index) => (
       <TagInfo data={tag} compact={false} key={index} />
     ));
   }
 
+  generateTagsSectionLabel(labelName: string) {
+    // TODO maybe add classes query-list-item-name column-name to label
+
+    return (
+      <label className="section-label">
+        <span className="section-title title-2">{labelName}</span>
+      </label>
+    );
+  }
+
+  generateBrowseTagLink() {
+    return (
+      <span>
+        <a className="browse-tags-link" href="/browse">
+          Browse all tags
+        </a>
+      </span>
+    );
+  }
+
+  sortTagsAlphabetically(tagArray: Tag[]) {
+    return tagArray.sort((a, b) => {
+      if (a.tag_name < b.tag_name) return -1;
+      if (a.tag_name > b.tag_name) return 1;
+      return 0;
+    });
+  }
+
   render() {
     const { isLoading, curatedTags, otherTags } = this.props;
 
     if (isLoading) {
       return <ShimmeringTagListLoader />;
     }
 
+    const filteredTags = otherTags.filter((tag) => {
```

Comment:
Should I filter out tags with usage of 0 in the Curated Tags section as well? Was hesitant because it affects previous behavior, on the upside it might help keeping curated tags less stale.